### PR TITLE
Update to Stylelint v16

### PR DESCRIPTION
**Pull request by MariaSolOs**
Updating dependencies to use Stylelint 16 (note that other ones got bumped too to satisfy peer deps).

A couple of rules got deprecated so I replaced them as recommended by the CLI:
- `scss/at-import-no-partial-leading-underscore` -> `scss/load-no-partial-leading-underscore`: They flag the same error but the later works for `@import`, `@use`, `@forward`, and `meta.load-css` `$url` parameters, while the first one only supports `@import`s.
- `scss/at-import-partial-extension-blacklist` -> `scss/at-import-partial-extension-disallowed-list`: [The rule got renamed to use inclusive language](https://github.com/stylelint-scss/stylelint-scss/pull/958).

I also noticed that there was a TODO comment which we can actually DO now 😸

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,13 +8,13 @@
...
-    "stylelint-config-standard": "^34.0.0",
+    "stylelint-config-standard": "^36.0.1",
     "stylelint-order": "^6.0.4"
   },
   "peerDependencies": {
-    "postcss": "^8.4.5",
-    "postcss-scss": "^4.0.2",
-    "stylelint": "^15.11.0"
+    "postcss": "^8.4.47",
+    "postcss-scss": "^4.0.9",
```
Comment (styu) [inline package.json:16]: technically these don't have to change, right? since for peer dependencies we probably want the widest possible range

Comment (MariaSolOs) [reply package.json:16]: Hmm not sure. At least for `postcss`, I set the version that `stylelint` indicates in their dependencies: https://github.com/stylelint/stylelint/blob/afbab11aacde5b0f59e3c755e2bd5fd4e5e320da/package.json#L157

Comment (MariaSolOs) [reply package.json:16]: @gluxon what are your thoughts on this?

Comment (styu) [reply package.json:16]: sorry forgot to reply - if that's the version stylelint needs than this makes sense!

**Review by styu — APPROVED** [review]

**Pull request merged**